source/OptimaJet.DWKit.StarterApplication/Repositories/Background/JobRepository.cs

```diff
@@ -27,17 +29,24 @@ public class JobRepository<TEntity, TId>
         protected readonly DbSet<TEntity> dbSet;
         protected readonly DbContext dbContext;
 
-        public JobRepository(IDbContextResolver contextResolver)
+        public StatusUpdateService StatusUpdateService { get; }
+
+        public JobRepository(
+            IDbContextResolver contextResolver,
+            StatusUpdateService statusUpdateService
+        )
         {
             this.dbContext = contextResolver.GetContext();
             this.dbSet = contextResolver.GetDbSet<TEntity>();
+            StatusUpdateService = statusUpdateService;
         }
 
         public virtual async Task<TEntity> CreateAsync(TEntity entity)
         {
             dbSet.Add(entity);
 
             var retval = await dbContext.SaveChangesAsync();
+            StatusUpdateService.OnInsert(entity as IStatusUpdate);
```

Comment:
I think status update service may be too specific of a name.

generally, what we want to capture any crud operation.
So, maybe I'd have something named: `EntityHooksService` with methods, like, `didCreate`, `didUpdate`, `didDestroy` -- past tense-ily named methods because the action already happened. 

Also, if we generally take any entity, instead of an `IStatusUpdate` and just define an entity as always defining a method that does Id.ToString(), then we can use this more a lot more than the current implementation.

But, for each of the hook methods, we'll probably want to pass a generic type as well, so if needed, we can have the EntityHooksService spin off background jobs based off the type of record (which would be beneficial of something is updating often, and would allow us to do things like keep the latest update, and drop all the yet-to-be-run notifications)

Reply:
@NullVoxPopuli: So if I change it to `EntityHookService` and `IHookEntity`?